presto-main/src/main/java/com/facebook/presto/operator/aggregation/GenericAccumulatorFactory.java

```diff
@@ -145,6 +146,76 @@ public GenericAccumulatorFactory(
         this.standaloneSpillerFactory = standaloneSpillerFactory;
     }
 
+    public static AccumulatorFactory generateAccumulatorFactory(
+                                          JavaAggregationFunctionImplementation functionImplementation,
+                                          List<Integer> argumentChannels,
+                                          Optional<Integer> maskChannel,
+                                          List<Type> sourceTypes,
+                                          List<Integer> orderByChannels,
+                                          List<SortOrder> orderings,
+                                          PagesIndex.Factory pagesIndexFactory,
+                                          boolean distinct,
+                                          JoinCompiler joinCompiler,
+                                          List<LambdaProvider> lambdaProviders,
+                                          boolean spillEnabled,
+                                          Session session,
+                                          StandaloneSpillerFactory standaloneSpillerFactory)
+    {
+        try {
+            Constructor<? extends Accumulator> accumulatorConstructor = functionImplementation.getAccumulatorClass().getConstructor(
+                    List.class,     /* List<AccumulatorStateDescriptor> stateDescriptors */
+                    List.class,     /* List<Integer> inputChannel */
+                    Optional.class, /* Optional<Integer> maskChannel */
+                    List.class      /* List<LambdaProvider> lambdaProviders */);
+
+            Constructor<? extends GroupedAccumulator> groupedAccumulatorConstructor = functionImplementation.getGroupedAccumulatorClass().getConstructor(
+                    List.class,     /* List<AccumulatorStateDescriptor> stateDescriptors */
+                    List.class,     /* List<Integer> inputChannel */
+                    Optional.class, /* Optional<Integer> maskChannel */
+                    List.class      /* List<LambdaProvider> lambdaProviders */);
+
+            return new GenericAccumulatorFactory(
+                    functionImplementation.getAggregationMetadata().getAccumulatorStateDescriptors(),
+                    accumulatorConstructor,
+                    groupedAccumulatorConstructor,
+                    lambdaProviders,
+                    argumentChannels,
+                    maskChannel,
+                    sourceTypes,
+                    orderByChannels,
+                    orderings,
+                    pagesIndexFactory,
+                    joinCompiler,
+                    session,
+                    distinct,
+                    spillEnabled,
+                    standaloneSpillerFactory);
+        }
+        catch (NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static AccumulatorFactory generateAccumulatorFactory(
```

Comment:
This one is probably only exist for testing purposes? Please annotate with `@VisibleForTesting` if that's the case.

Reply:
the methos is also used in `AggregateWindowFunction` class and `presto-benchmark` module